src/saltext/vmware/modules/vmc_distributed_firewall_rules.py

```diff
@@ -0,0 +1,118 @@
+"""
+Salt execution module for VMC distributed firewall rules
+Provides methods to Create, Read, Update and Delete Distributed firewall rules.
+"""
+import logging
+import os
+
+from saltext.vmware.utils import vmc_constants
+from saltext.vmware.utils import vmc_request
+from saltext.vmware.utils import vmc_templates
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = "vmc_distributed_firewall_rules"
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def get(
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    domain_id,
+    security_policy_id,
+    verify_ssl=True,
+    cert=None,
+    cursor=None,
+    page_size=None,
+    sort_by=None,
+    sort_ascending=None,
+):
+    """
+    Retrieves distributed firewall rules for Given SDDC
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt vm_minion vmc_distributed_firewall_rules.get hostname=nsxt-manager.local domain_id=cgw ...
+
+    hostname
+        The host name of NSX-T manager
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console
+
+    org_id
+        The Id of organization to which the SDDC belongs to
+
+    sddc_id
+        The Id of SDDC for which the distributed firewall rules should be retrieved
+
+    domain_id
+        The domain_id for which the distributed firewall rules should be retrieved
+
+    security_policy_id
+        Security policy id for which rule should be retrieved
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL certificate file to connect to NSX-T manager.
+        The certificate can be retrieved from browser.
+
+    cursor
+        (Optional) Opaque cursor to be used for getting next page of records (supplied by current result page)
+
+    page_size
+        (Optional) Maximum number of results to return in this page. Default page size is 1000.
+
+    sort_by
+        (Optional) Field by which records are sorted
+
+    sort_ascending
+        (Optional) Boolean value to sort result in ascending order. Enabled by default.
+
+    """
+    log.info("Retrieving %s distributed firewall rules for SDDC %s", domain_id, sddc_id)
+
+    api_url_base = vmc_request.set_base_url(hostname)
+    api_url = (
+        "{base_url}vmc/reverse-proxy/api/orgs/{org_id}/sddcs/{sddc_id}/"
+        "policy/api/v1/infra/domains/{domain_id}/security-policies/{security_policy_id}/rules"
+    )
+    api_url = api_url.format(
+        base_url=api_url_base,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_policy_id=security_policy_id,
+    )
+
+    params = vmc_request._filter_kwargs(
+        allowed_kwargs=["cursor", "page_size", "sort_ascending", "sort_by"],
```

Comment:
**suggestion** looks like this PR is missing tests to ensure the params are correct. After this line I was able to replace params with `params = None` and all the tests continue to pass.

Reply:
Hi @waynew , we agree with this suggestion but as we already done the implementation through out, would be difficult to address this for now. we can surely take this as further test improvement later.